gtk4/src/expression.rs

```diff
@@ -0,0 +1,362 @@
+        notify: F,
+    ) -> ExpressionWatch {
+        assert_initialized_main_thread!();
+        unsafe extern "C" fn notify_trampoline<F: Fn() + 'static>(user_data: glib::ffi::gpointer) {
+            let f: &F = &*(user_data as *const F);
+            f()
+        }
+        unsafe extern "C" fn destroy_func<F: Fn() + 'static>(user_data: glib::ffi::gpointer) {
+            let _callback: Box_<Option<Box_<F>>> = Box_::from_raw(user_data as *mut _);
+        }
+        let callback_data: Box_<F> = Box_::new(notify);
+        unsafe {
+            let ret = ffi::gtk_expression_watch(
+                self.to_glib_none().0,
+                this.map(|t| t.as_ref()).to_glib_none().0,
+                Some(notify_trampoline::<F> as _),
+                Box_::into_raw(callback_data) as *mut _,
+                Some(destroy_func::<F> as _),
+            );
+            ExpressionWatch(*ret)
+        }
+    }
+}
+
+impl<'a> glib::value::FromValueOptional<'a> for Expression {
+    unsafe fn from_value_optional(value: &'a Value) -> Option<Self> {
+        from_glib_full(ffi::gtk_value_dup_expression(value.to_glib_none().0))
+    }
+}
+
+impl glib::value::SetValue for Expression {
+    unsafe fn set_value(value: &mut Value, this: &Self) {
+        ffi::gtk_value_set_expression(value.to_glib_none_mut().0, this.to_glib_none().0)
+    }
+}
+
+impl glib::value::SetValueOptional for Expression {
+    unsafe fn set_value_optional(value: &mut glib::Value, this: Option<&Self>) {
+        ffi::gtk_value_set_expression(value.to_glib_none_mut().0, this.to_glib_none().0)
+    }
+}
+
+/// Register Expression's ParamSpec support
+trait GtkParamSpecExt {
+    fn expression(name: &str, nick: &str, blurb: &str, flags: glib::ParamFlags) -> Self;
+}
+
+impl GtkParamSpecExt for glib::ParamSpec {
+    #[doc(alias = "gtk_param_spec_expression")]
+    fn expression(name: &str, nick: &str, blurb: &str, flags: glib::ParamFlags) -> Self {
+        skip_assert_initialized!();
+        unsafe {
+            from_glib_full(ffi::gtk_param_spec_expression(
+                name.to_glib_none().0,
+                nick.to_glib_none().0,
+                blurb.to_glib_none().0,
+                flags.to_glib(),
+            ))
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct ExpressionWatch(ffi::GtkExpressionWatch);
+
+impl ExpressionWatch {
+    #[doc(alias = "gtk_expression_watch_evaluate")]
+    pub fn evaluate(&self) -> Option<Value> {
+        skip_assert_initialized!();
+        unsafe {
+            let mut value = Value::uninitialized();
+            let ret =
+                ffi::gtk_expression_watch_evaluate(self.0 as *mut _, value.to_glib_none_mut().0);
+            if from_glib(ret) {
+                Some(value)
+            } else {
+                None
+            }
+        }
+    }
+
+    #[doc(alias = "gtk_expression_watch_unwatch")]
+    pub fn unwatch(&self) {
+        unsafe { ffi::gtk_expression_watch_unwatch(self.0 as *mut _) }
+    }
+}
+
+#[macro_export]
```

Comment:
Wouldn't this make the macro visible outside the crate too? We probably don't want that or are there expressions defined outside GTK (or is that possible)?

Reply:
I don't think we want that macro to be visible, no. Removed it